### ISSUE TITLE
How do I set up the SAIC/MQTT gateway correctly? - OpenWB Software 2 + Synology Docker Container

### ISSUE THREAD

**Issue by klefreak** [SAIC-iSmart-API/saic-python-mqtt-gateway#167]
Hello:
I allready used the Saic-Gateway for over a year with OpenWB Software 1.9 + Synology Docker.
it worked good using the old OPENWB_LP_MAP Command

Since update on Software 2 on my open WB it does not work any more.

I Created the needed charging Station.json -->

`"chargeStateTopic": "openWB/lp/1/boolChargeStat",
        "chargingValue": "1",
        "socTopic": "openWB/set/vehicle/6/get/soc",
        "chargerConnectedTopic": "openWB/lp/1/boolPlugStat",
        "chargerConnectedValue": "1",
        "vin": "LSJEXXXXXXXXXXXXX"`

I'm unsure at "charge state topic", "ChargingValue", "chargerConnectedTopic" because none of these Vales are "suggested in OpenWB Software..
the only Thing i can find is "socTopic" that is adaptet to the needed "6" vehicle...

how can i find out if or how to set the other parameters..

second thing:
i'm not shure if my car is read out correctly by the Gateway ... see Log posted here:
https://forum.openwb.de/viewtopic.php?p=102499#p102499 

thanks for clarification :)

ps: im a noob in using these things - sorry

**Comment by nanomad** [comment]
Hi,
What version of the MQTT gateway are you running?
it looks like the car is not communicating properly with the MQTT gateway, here's what i would do:
- Stop the current gateway
- "Wake the car up" by moving a couple of meters
- Re-start the gateway

After you do that feel free to post the log file so we can analyze if it is working properly or not.

**Comment by klefreak** [comment]
MQTT gateway is from :Latest -->were can i find actual version number?

moved Car
%SOC Status via App works properly... 2.0.2
restarted Container in "Debug"
[saicismartapi-saic-python-mqtt-gateway.txt](https://github.com/SAIC-iSmart-API/saic-python-mqtt-gateway/files/14284966/saicismartapi-saic-python-mqtt-gateway.txt)

looks like connection to SAIC failed ???

**Comment by tosate** [comment]
There is actually a sample file for openWB in the repository. Have a look at charging-stations.json.sample.

Your file should look like this:
```
[
    {
        "chargeStateTopic": "openWB/lp/1/boolChargeStat",
        "chargingValue": "1",
        "socTopic": "openWB/set/lp/1/%Soc",
        "chargerConnectedTopic": "openWB/lp/1/boolPlugStat",
        "chargerConnectedValue": "1",
        "vin": "vin1"
    }
]
```

Seems like the MQTT topic structure has changed in openWB software version 2.0.
I have found [a discussion in German about this topic](https://forum.openwb.de/viewtopic.php?t=7785)

Seems like `openWB/lp/` became `openWB/chargepoint`
I don't know about the rest of the topic since it is not visible on the screenshot in the thread. But as they are suggesting in the forum, you could use MQTT Explorer to find it out.

**Comment by klefreak** [comment]
@tosate 
thanks for the hint..

tried to find the topics...
looks like this as far as i can interpret with my little knowledge

`"chargeStateTopic": "openWB/internal_chargepoint/0/get/charge_state",
"chargingValue": "true",
"socTopic": "openWB/set/vehicle/6/get/soc",
"chargerConnectedTopic": "openWB/internal_chargepoint/0/get/plug_state",
"chargerConnectedValue": "true",
"vin": "LSJExxxxxxxxxxx"` maybe someone can confirm or "repair" if my findings are broken

**Comment by tosate** [comment]
Hi @klefreak,

I have updated my openWB charging station to version 2.0 on the weekend.
There is a default car with ID 0 that cannot be modified. Therefore, I have created an additional car that has received the ID 2. This explains why there is the 2 in my MQTT topics:
```
[
    {
        "chargeStateTopic": "openWB/chargepoint/2/get/charge_state",
        "chargingValue": "true",
        "socTopic": "openWB/set/vehicle/2/get/soc",
        "rangeTopic": "openWB/set/vehicle/2/get/range"
        "chargerConnectedTopic": "openWB/chargepoint/2/get/plug_state",
        "chargerConnectedValue": "true",
        "vin": "vin1"
    }
]
```

Providing the SoC to the openWB is working.
I am still testing if the charging is recognized by the gateway.

**Comment by klefreak** [comment]
Update:
i'm testing this Charging Station File:
`"chargeStateTopic": "openWB/chargepoint/4/get/charge_state",
        "chargingValue": "true",
        "socTopic": "openWB/set/vehicle/6/get/soc",
        "rangeTopic": "openWB/set/vehicle/6/get/range"
        "chargerConnectedTopic": "openWB/chargepoint/4/get/plug_state",
        "chargerConnectedValue": "true",
        "vin": "VIN1"`
My Charging Point got ID4 and My Car got ID6 in the Box...

Gateway writes This: (but no SOC UPdate in the Box - What is wrong??

`<!DOCTYPE html>
<html><head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<style>body{font-family:"Lucida Console",Consola,Monaco,Monospace,"Courier New",Courier,Serif;}h2{text-align:center;}table{width:95%;border-collapse:collapse;}table,th,td{border:1px solid black;}td{white-space:nowrap;text-align:left;}</style>
</head>
<body>
<h2>saicismartapi-saic-python-mqtt-gateway-1</h2>
<table>
<tr><th>date</th><th>stream</th><th>content</th></tr>
<tr><td>2024/02/22&nbsp;21:01:03</td><td>stdout</td><td>2024-02-22&nbsp;20:01:03,762&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;Job&nbsp;"Check&nbsp;for&nbsp;new&nbsp;messages&nbsp;(trigger:&nbsp;interval[0:01:00],&nbsp;next&nbsp;run&nbsp;at:&nbsp;2024-02-22&nbsp;20:02:03&nbsp;UTC)"&nbsp;executed&nbsp;successfully&nbsp;-&nbsp;apscheduler.executors.default
</td></tr>
<tr><td>2024/02/22&nbsp;21:01:03</td><td>stdout</td><td>2024-02-22&nbsp;20:01:03,761&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;MessageResp(alarmNumber=0,&nbsp;commandNumber=0,&nbsp;messages=[],&nbsp;newsNumber=0,&nbsp;recordsNumber=None,&nbsp;totalNumber=0)&nbsp;unread&nbsp;messages&nbsp;-&nbsp;__main__
</td></tr>
<tr><td>2024/02/22&nbsp;21:01:03</td><td>stdout</td><td>2024-02-22&nbsp;20:01:03,571&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;Running&nbsp;job&nbsp;"Check&nbsp;for&nbsp;new&nbsp;messages&nbsp;(trigger:&nbsp;interval[0:01:00],&nbsp;next&nbsp;run&nbsp;at:&nbsp;2024-02-22&nbsp;20:02:03&nbsp;UTC)"&nbsp;(scheduled&nbsp;at&nbsp;2024-02-22&nbsp;20:01:03.570039+00:00)&nbsp;-&nbsp;apscheduler.executors.default
</td></tr>
<tr><td>2024/02/22&nbsp;21:00:45</td><td>stdout</td><td>2024-02-22&nbsp;20:00:45,084&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;Refreshing&nbsp;ABRP&nbsp;status&nbsp;succeeded...&nbsp;-&nbsp;__main__
</td></tr>
<tr><td>2024/02/22&nbsp;21:00:45</td><td>stdout</td><td>2024-02-22&nbsp;20:00:45,084&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;Refreshing&nbsp;vehicle&nbsp;status&nbsp;succeeded...&nbsp;-&nbsp;__main__
</td></tr>
<tr><td>2024/02/22&nbsp;21:00:45</td><td>stdout</td><td>2024-02-22&nbsp;20:00:45,037&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;Updating&nbsp;scheduled&nbsp;battery&nbsp;heating&nbsp;status&nbsp;-&nbsp;__main__
</td></tr>
<tr><td>2024/02/22&nbsp;21:00:45</td><td>stdout</td><td>2024-02-22&nbsp;20:00:45,036&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;Setting&nbsp;charging&nbsp;query&nbsp;interval&nbsp;in&nbsp;vehicle&nbsp;handler&nbsp;for&nbsp;VIN&nbsp;&nbsp;to&nbsp;0:00:00&nbsp;-&nbsp;vehicle
</td></tr>
<tr><td>2024/02/22&nbsp;21:00:41</td><td>stdout</td><td>2024-02-22&nbsp;20:00:41,358&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;Updating&nbsp;charging&nbsp;status&nbsp;-&nbsp;__main__
</td></tr>
<tr><td>2024/02/22&nbsp;21:00:30</td><td>stdout</td><td>2024-02-22&nbsp;20:00:30,954&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;Updating&nbsp;vehicle&nbsp;status&nbsp;-&nbsp;__main__
</td></tr>
<tr><td>2024/02/22&nbsp;21:00:03</td><td>stdout</td><td>2024-02-22&nbsp;20:00:03,753&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;Job&nbsp;"Check&nbsp;for&nbsp;new&nbsp;messages&nbsp;(trigger:&nbsp;interval[0:01:00],&nbsp;next&nbsp;run&nbsp;at:&nbsp;2024-02-22&nbsp;20:01:03&nbsp;UTC)"&nbsp;executed&nbsp;successfully&nbsp;-&nbsp;apscheduler.executors.default
</td></tr>
<tr><td>2024/02/22&nbsp;21:00:03</td><td>stdout</td><td>2024-02-22&nbsp;20:00:03,753&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;MessageResp(alarmNumber=0,&nbsp;commandNumber=0,&nbsp;messages=[],&nbsp;newsNumber=0,&nbsp;recordsNumber=None,&nbsp;totalNumber=0)&nbsp;unread&nbsp;messages&nbsp;-&nbsp;__main__
</td></tr>
<tr><td>2024/02/22&nbsp;21:00:03</td><td>stdout</td><td>2024-02-22&nbsp;20:00:03,571&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;Running&nbsp;job&nbsp;"Check&nbsp;for&nbsp;new&nbsp;messages&nbsp;(trigger:&nbsp;interval[0:01:00],&nbsp;next&nbsp;run&nbsp;at:&nbsp;2024-02-22&nbsp;20:01:03&nbsp;UTC)"&nbsp;(scheduled&nbsp;at&nbsp;2024-02-22&nbsp;20:00:03.570039+00:00)&nbsp;-&nbsp;apscheduler.executors.default
</td></tr>
<tr><td>2024/02/22&nbsp;20:59:03</td><td>stdout</td><td>2024-02-22&nbsp;19:59:03,785&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;Job&nbsp;"Check&nbsp;for&nbsp;new&nbsp;messages&nbsp;(trigger:&nbsp;interval[0:01:00],&nbsp;next&nbsp;run&nbsp;at:&nbsp;2024-02-22&nbsp;20:00:03&nbsp;UTC)"&nbsp;executed&nbsp;successfully&nbsp;-&nbsp;apscheduler.executors.default
</td></tr>
<tr><td>2024/02/22&nbsp;20:59:03</td><td>stdout</td><td>2024-02-22&nbsp;19:59:03,785&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;MessageResp(alarmNumber=0,&nbsp;commandNumber=0,&nbsp;messages=[],&nbsp;newsNumber=0,&nbsp;recordsNumber=None,&nbsp;totalNumber=0)&nbsp;unread&nbsp;messages&nbsp;-&nbsp;__main__
</td></tr>
<tr><td>2024/02/22&nbsp;20:59:03</td><td>stdout</td><td>2024-02-22&nbsp;19:59:03,572&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;Running&nbsp;job&nbsp;"Check&nbsp;for&nbsp;new&nbsp;messages&nbsp;(trigger:&nbsp;interval[0:01:00],&nbsp;next&nbsp;run&nbsp;at:&nbsp;2024-02-22&nbsp;20:00:03&nbsp;UTC)"&nbsp;(scheduled&nbsp;at&nbsp;2024-02-22&nbsp;19:59:03.570039+00:00)&nbsp;-&nbsp;apscheduler.executors.default
</td></tr>
<tr><td>2024/02/22&nbsp;20:58:30</td><td>stdout</td><td>2024-02-22&nbsp;19:58:30,806&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;Refreshing&nbsp;ABRP&nbsp;status&nbsp;succeeded...&nbsp;-&nbsp;__main__
</td></tr>
<tr><td>2024/02/22&nbsp;20:58:30</td><td>stdout</td><td>2024-02-22&nbsp;19:58:30,806&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;Refreshing&nbsp;vehicle&nbsp;status&nbsp;succeeded...&nbsp;-&nbsp;__main__
</td></tr>
<tr><td>2024/02/22&nbsp;20:58:30</td><td>stdout</td><td>2024-02-22&nbsp;19:58:30,757&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;Updating&nbsp;scheduled&nbsp;battery&nbsp;heating&nbsp;status&nbsp;-&nbsp;__main__
</td></tr>
<tr><td>2024/02/22&nbsp;20:58:30</td><td>stdout</td><td>2024-02-22&nbsp;19:58:30,757&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;Setting&nbsp;charging&nbsp;query&nbsp;interval&nbsp;in&nbsp;vehicle&nbsp;handler&nbsp;for&nbsp;VIN&nbsp;&nbsp;to&nbsp;0:00:00&nbsp;-&nbsp;vehicle
</td></tr>
<tr><td>2024/02/22&nbsp;20:58:27</td><td>stdout</td><td>2024-02-22&nbsp;19:58:27,084&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;Updating&nbsp;charging&nbsp;status&nbsp;-&nbsp;__main__
</td></tr>
<tr><td>2024/02/22&nbsp;20:58:13</td><td>stdout</td><td>2024-02-22&nbsp;19:58:13,585&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;Updating&nbsp;vehicle&nbsp;status&nbsp;-&nbsp;__main__
</td></tr>
<tr><td>2024/02/22&nbsp;20:58:13</td><td>stdout</td><td>2024-02-22&nbsp;19:58:13,585&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;Setting&nbsp;refresh&nbsp;mode&nbsp;to&nbsp;periodic&nbsp;-&nbsp;vehicle
</td></tr>
<tr><td>2024/02/22&nbsp;20:58:13</td><td>stdout</td><td>2024-02-22&nbsp;19:58:13,585&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;Setting&nbsp;charging&nbsp;query&nbsp;interval&nbsp;in&nbsp;vehicle&nbsp;handler&nbsp;for&nbsp;VIN&nbsp;xxxxxxx&nbsp;to&nbsp;0:00:00&nbsp;-&nbsp;vehicle
</td></tr>
<tr><td>2024/02/22&nbsp;20:58:13</td><td>stdout</td><td>2024-02-22&nbsp;19:58:13,585&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;Setting&nbsp;inactive&nbsp;query&nbsp;interval&nbsp;in&nbsp;vehicle&nbsp;handler&nbsp;for&nbsp;VIN&nbsp;xxxxxxx&nbsp;to&nbsp;1&nbsp;day,&nbsp;0:00:00&nbsp;-&nbsp;vehicle
</td></tr>
<tr><td>2024/02/22&nbsp;20:58:13</td><td>stdout</td><td>2024-02-22&nbsp;19:58:13,585&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;Setting&nbsp;after&nbsp;shutdown&nbsp;query&nbsp;interval&nbsp;in&nbsp;vehicle&nbsp;handler&nbsp;for&nbsp;VIN&nbsp;xxxxxxx&nbsp;to&nbsp;0:02:00&nbsp;-&nbsp;vehicle
</td></tr>
<tr><td>2024/02/22&nbsp;20:58:13</td><td>stdout</td><td>2024-02-22&nbsp;19:58:13,585&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;Setting&nbsp;charging&nbsp;query&nbsp;interval&nbsp;in&nbsp;vehicle&nbsp;handler&nbsp;for&nbsp;VIN&nbsp;xxxxxxx&nbsp;to&nbsp;0:00:00&nbsp;-&nbsp;vehicle
</td></tr>
<tr><td>2024/02/22&nbsp;20:58:13</td><td>stdout</td><td>2024-02-22&nbsp;19:58:13,584&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;Setting&nbsp;active&nbsp;query&nbsp;interval&nbsp;in&nbsp;vehicle&nbsp;handler&nbsp;for&nbsp;VIN&nbsp;xxxxxxxxxx&nbsp;to&nbsp;0:00:30&nbsp;-&nbsp;vehicle
</td></tr>
<tr><td>2024/02/22&nbsp;20:58:03</td><td>stdout</td><td>2024-02-22&nbsp;19:58:03,570&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;Added&nbsp;job&nbsp;"Check&nbsp;for&nbsp;new&nbsp;messages"&nbsp;to&nbsp;job&nbsp;store&nbsp;"default"&nbsp;-&nbsp;apscheduler.scheduler
</td></tr>
<tr><td>2024/02/22&nbsp;20:58:03</td><td>stdout</td><td>2024-02-22&nbsp;19:58:03,516&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;Registering&nbsp;for&nbsp;['ALARM_TYPE_VEHICLE_FAULT',&nbsp;'ALARM_TYPE_GEOFENCE',&nbsp;'ALARM_TYPE_VEHICLE_START']&nbsp;messages.&nbsp;vin=xxxxxxxxxxx&nbsp;-&nbsp;__main__
</td></tr>
<tr><td>2024/02/22&nbsp;20:58:03</td><td>stdout</td><td>2024-02-22&nbsp;19:58:03,227&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;Fetching&nbsp;vehicle&nbsp;list&nbsp;-&nbsp;__main__
</td></tr>
<tr><td>2024/02/22&nbsp;20:58:03</td><td>stdout</td><td>2024-02-22&nbsp;19:58:03,227&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;Logged&nbsp;in&nbsp;as&nbsp;6602599939&nbsp;-&nbsp;__main__
</td></tr>
<tr><td>2024/02/22&nbsp;20:58:02</td><td>stdout</td><td>2024-02-22&nbsp;19:58:02,777&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;[SUBACK]&nbsp;4&nbsp;(0,)&nbsp;-&nbsp;gmqtt
</td></tr>
<tr><td>2024/02/22&nbsp;20:58:02</td><td>stdout</td><td>2024-02-22&nbsp;19:58:02,777&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;[SUBACK]&nbsp;3&nbsp;(0,)&nbsp;-&nbsp;gmqtt
</td></tr>
<tr><td>2024/02/22&nbsp;20:58:02</td><td>stdout</td><td>2024-02-22&nbsp;19:58:02,777&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;[SUBACK]&nbsp;2&nbsp;(0,)&nbsp;-&nbsp;gmqtt
</td></tr>
<tr><td>2024/02/22&nbsp;20:58:02</td><td>stdout</td><td>2024-02-22&nbsp;19:58:02,776&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;[SUBACK]&nbsp;1&nbsp;(0,)&nbsp;-&nbsp;gmqtt
</td></tr>
<tr><td>2024/02/22&nbsp;20:58:02</td><td>stdout</td><td>2024-02-22&nbsp;19:58:02,767&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;Logging&nbsp;in&nbsp;to&nbsp;SAIC&nbsp;API&nbsp;-&nbsp;__main__
</td></tr>
<tr><td>2024/02/22&nbsp;20:58:02</td><td>stdout</td><td>2024-02-22&nbsp;19:58:02,767&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;Scheduler&nbsp;started&nbsp;-&nbsp;apscheduler.scheduler
</td></tr>
<tr><td>2024/02/22&nbsp;20:58:02</td><td>stdout</td><td>2024-02-22&nbsp;19:58:02,762&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;[SEND&nbsp;SUB]&nbsp;4&nbsp;[b'saic/6602599939/vehicles/+/refresh/period/+/set']&nbsp;-&nbsp;gmqtt.mqtt.package
</td></tr>
<tr><td>2024/02/22&nbsp;20:58:02</td><td>stdout</td><td>2024-02-22&nbsp;19:58:02,762&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;[SEND&nbsp;SUB]&nbsp;3&nbsp;[b'saic/6602599939/vehicles/+/refresh/mode/set']&nbsp;-&nbsp;gmqtt.mqtt.package
</td></tr>
<tr><td>2024/02/22&nbsp;20:58:02</td><td>stdout</td><td>2024-02-22&nbsp;19:58:02,762&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;[SEND&nbsp;SUB]&nbsp;2&nbsp;[b'saic/6602599939/vehicles/+/+/+/+/set']&nbsp;-&nbsp;gmqtt.mqtt.package
</td></tr>
<tr><td>2024/02/22&nbsp;20:58:02</td><td>stdout</td><td>2024-02-22&nbsp;19:58:02,762&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;[SEND&nbsp;SUB]&nbsp;1&nbsp;[b'saic/6602599939/vehicles/+/+/+/set']&nbsp;-&nbsp;gmqtt.mqtt.package
</td></tr>
<tr><td>2024/02/22&nbsp;20:58:02</td><td>stdout</td><td>2024-02-22&nbsp;19:58:02,762&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;Connected&nbsp;to&nbsp;MQTT&nbsp;broker&nbsp;-&nbsp;mqtt_publisher
</td></tr>
<tr><td>2024/02/22&nbsp;20:58:02</td><td>stdout</td><td>2024-02-22&nbsp;19:58:02,759&nbsp;[&nbsp;&nbsp;INFO&nbsp;&nbsp;]&nbsp;[CONNECTION&nbsp;MADE]&nbsp;-&nbsp;gmqtt.mqtt.protocol
</td></tr>
</table></body></html>`

**Comment by tosate** [comment]
Hi @klefreak,

go to configuration > vehicles (Konfiguration > Fahrzeuge) in your openWB.
Select SoC-Modul: MQTT

The box shows you the SoC and the range MQTT topic.#
![image](https://github.com/SAIC-iSmart-API/saic-python-mqtt-gateway/assets/35031183/347fc6c3-64af-41ae-b88e-ca7de45e5565)

Your charge_state topic contains the ID 4
Your SoC topic contains the ID 6

The topics for the same car should have the same ID.

**Comment by klefreak** [comment]
screenshot from openWB
![grafik](https://github.com/SAIC-iSmart-API/saic-python-mqtt-gateway/assets/131512515/10d0703a-e2a5-4119-98fd-586e25834f3f)

Screenshot from MQTT Expl.
![grafik](https://github.com/SAIC-iSmart-API/saic-python-mqtt-gateway/assets/131512515/8bf8f906-306f-4d84-9bdb-2633d1623ac5)


have to wait if it works as my wife uses the car :D

**Comment by klefreak** [comment]
Hello @tosate 

Thx for your Time
i tried to ad a second car.. got ID2 and copied your File-configuration as now i also use car ID2 and charging id2..
MQTT is set up..
under mqtt Explorer i can read SOC status in SAIC-Tree --> 
![grafik](https://github.com/SAIC-iSmart-API/saic-python-mqtt-gateway/assets/131512515/024e21d3-4443-4e28-a598-4fea5b3947c4)

but it is not written into the car.. i dont know what to do else..
feels like i make som stupid mistakes??

@tosate want to jump in via ANYDESK ??
(i'm in Austria, so german would be possible also phone would be possible :)

**Comment by tosate** [comment]
Hi @klefreak,

first of all, sorry for the late response.
I think the gateway does not pick up your `charging-stations.json` file.
By default, the release comes with a file that contains an empty array `[]`, which has no effect.
If you are running it as a docker container, you need to map your custom file into the running container.

There is an example docker-compose.yml file in the repository. It contains the following mapping:

```
[...]
    volumes:
      - ./charging-stations.json:/usr/src/app/charging-stations.json
```

This maps the `charging-stations.json` file from the current directory into the container and replaces the file `/usr/src/app/charging-stations.json`.

I am not familiar with the Synology Container Manager. However, I think that the mapping is done under "Volume-Einstellungen".
What you have configured looks like:

```
[...]
    volumes:
      - /docker/saic/charging-stations.json:/charging-stations.json
```

This maps your `charging-stations.json` file into the root directory of the container instead of `/usr/src/app/`

So long story short: Please try with `/usr/src/app/charging-stations.json` in your mapping.

**Comment by klefreak** [comment]
@tosate 
You are right !

solved it:
1. i removed the Environment-Variable
2. i adopted the File Mapping in Docker Container like you suggested..
3. i recreated the charging-stations.json because in "my" used version i got errors..
4. --> start container --> works  :)

Thank you for your Help!
i will make a summary for OpenWB-Forum in relating SAIC-Thread... (in a view days...)
greetings Klemens
